lib/PGUtil.pm

```diff
@@ -163,4 +164,51 @@ sub pretty_print_tex {
 		$out;
 }
 
+sub pretty_print_text { 
+	my $r_input = shift;
+	my $level   = shift;
+	my $indent = 4;   #FIXME -- should we allow this to be modified somehow?
+	
+	$level--;
+	return "PGalias has too much info. Try \\\$PG->{PG\\_alias}->{resource\\_list}" if ref($r_input) eq 'PGalias';  # PGalias just has too much information
+	return 'too deep' unless $level>0;  #only print four levels of hashes (safety feature)
+
+	my $out = "";
+	my $type = ref($r_input);
+
+	if (defined($type) and $type) {
+		$out .= " type = $type; ";
+	} elsif (! defined($r_input )) {
+		$out .= " type = UNDEFINED; ";
+	}
+	return $out." " unless defined($r_input);
+	
+	if ( ref($r_input) =~/HASH/ or "$r_input" =~/HASH/ ) {
+	    $out .= "{\n";
+	    $indent++;
+ 		foreach my $key (sort keys %{$r_input})  {
+ 			$out .= "  "x$indent."$key => " . pretty_print_text( $r_input->{$key}, $indent ) . "\n";
+ 		}
```

Comment:
You're right.  I could have sworn I did a global replace on this subroutine when I transferred it from renderProblem.pl -- but I guess not.  Thanks for catching it. 

This will be useful for printing things to log files but I think the text version of pretty_print is going to be most useful on the WW2 side of the applications.  There are several versions of it in the code already and in my opinion they should be consolidated.